In [14]:
import pandas as pd

df = pd.read_json("carrera_san_silvestre.json")

df.head()


runner_name         finish_time age_group gender  \
0  JOSE MANUEL VIEITO VILACOBA 2026-02-11 00:23:50      None      M   
1     JOSE ANTONIO ABUIN TREUS 2026-02-11 00:28:26      None      M   
2           PABLO LOPEZ FRANCO 2026-02-11 00:28:37      None      M   
3      FELIX QUINTANS MELENDEZ 2026-02-11 00:28:51      None      M   
4         DAVID ROMERO ALBORÉS 2026-02-11 00:28:54      None      M   

   race_date  location            race_distance  
0       2010  A Coruña  Distancia no encontrada  
1       2010  A Coruña  Distancia no encontrada  
2       2010  A Coruña  Distancia no encontrada  
3       2010  A Coruña  Distancia no encontrada  
4       2010  A Coruña  Distancia no encontrada

In [15]:
import pandas as pd
import os
os.getcwd()




'/home/paula/RaceAnalytics/proyecto_sansilvestre/proyecto_sansilvestre'

In [16]:
df.info()
df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 39862 entries, 0 to 39861
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   runner_name    39862 non-null  object        
 1   finish_time    39862 non-null  datetime64[ns]
 2   age_group      37388 non-null  object        
 3   gender         39847 non-null  object        
 4   race_date      39862 non-null  int64         
 5   location       39862 non-null  object        
 6   race_distance  39862 non-null  object        
dtypes: datetime64[ns](1), int64(1), object(5)
memory usage: 2.1+ MB


runner_name         finish_time age_group gender  \
0  JOSE MANUEL VIEITO VILACOBA 2026-02-11 00:23:50      None      M   
1     JOSE ANTONIO ABUIN TREUS 2026-02-11 00:28:26      None      M   
2           PABLO LOPEZ FRANCO 2026-02-11 00:28:37      None      M   
3      FELIX QUINTANS MELENDEZ 2026-02-11 00:28:51      None      M   
4         DAVID ROMERO ALBORÉS 2026-02-11 00:28:54      None      M   

   race_date  location            race_distance  
0       2010  A Coruña  Distancia no encontrada  
1       2010  A Coruña  Distancia no encontrada  
2       2010  A Coruña  Distancia no encontrada  
3       2010  A Coruña  Distancia no encontrada  
4       2010  A Coruña  Distancia no encontrada

# Convertir tiempos a segundos

In [17]:
# 1. Aseguramos que sea string y extraemos solo la parte del tiempo (HH:MM:SS)
df['finish_time'] = df['finish_time'].astype(str).str.split(' ').str[-1]

# 2. Convertimos a Timedelta para que Pandas entienda que es una duración
df['finish_time'] = pd.to_timedelta(df['finish_time'])

# 3. Calculamos los segundos totales
df["finish_seconds"] = df["finish_time"].dt.total_seconds()

# 4. Aseguramos que el año sea numérico
df["year"] = pd.to_numeric(df["race_date"])

# Mostramos el resultado (Corregido: eliminada la coma extra y añadido runner_name para contexto)
print(df[["runner_name", "finish_time", "finish_seconds", "year"]].head())

                   runner_name     finish_time  finish_seconds  year
0  JOSE MANUEL VIEITO VILACOBA 0 days 00:23:50          1430.0  2010
1     JOSE ANTONIO ABUIN TREUS 0 days 00:28:26          1706.0  2010
2           PABLO LOPEZ FRANCO 0 days 00:28:37          1717.0  2010
3      FELIX QUINTANS MELENDEZ 0 days 00:28:51          1731.0  2010
4         DAVID ROMERO ALBORÉS 0 days 00:28:54          1734.0  2010


# Performance Metrics 

Average finish times by age group 

In [26]:
avg_time_by_age = df.groupby("age_group")["finish_seconds"].mean().sort_values(ascending=True)
avg_time_by_age

age_group
SENIOR MASCULINO     2049.473485
VETERANOS B MASC     2061.259174
PROMESA MASCULINO    2062.615385
VETERANOS A MASC     2073.408392
VETERANOS C MASC     2143.990566
JVM                  2166.788945
VETERANOS D MASC     2198.880000
PROMESA FEMENINA     2342.437500
JV2M                 2376.987044
VETERANAS B FEM      2409.958333
VTAM                 2410.250288
VETERANOS D FEM      2460.333333
VETERANAS A FEM      2461.515432
VETERANAS C FEM      2462.095238
SNM                  2464.085891
VTBM                 2495.097778
SENIOR FEMENINA      2500.747664
JVF                  2570.211765
JV1M                 2609.476845
VTCM                 2611.531677
DVISM                2691.133333
VTDM                 2787.295026
SILLAM               2790.555556
VTAF                 2926.704309
SILLAF               2972.800000
SNF                  2980.284140
VTBF                 3020.152812
JV2F                 3052.546748
JV1F                 3122.881201
DINTM                3187.333333


Average finish times by gender 

In [19]:
avg_time_by_gender = df.groupby("gender")["finish_seconds"].mean()
avg_time_by_gender


gender
F    2982.888980
M    2444.773107
Name: finish_seconds, dtype: float64

# Trend Analysis 

In [20]:
# Extraer el año
df["year"] = pd.to_numeric(df["race_date"], errors='coerce')

Participation evolution over the years 

In [21]:
participants_by_year = df.groupby("year").size()
participants_by_year


year
2010     460
2011     895
2012    1119
2014    2200
2015    2697
2016    3119
2017    3341
2018    3801
2019    4064
2021    2610
2023    5817
2024    4373
2025    5366
dtype: int64

Male/female participation trends 

In [22]:
gender_participation  = df.groupby(["year", "gender"]).size().unstack()
gender_participation

gender       F       M
year                  
2010       NaN   460.0
2011       NaN   895.0
2012       NaN  1119.0
2014     615.0  1585.0
2015     851.0  1846.0
2016    1006.0  2113.0
2017    1131.0  2210.0
2018    1322.0  2479.0
2019    1448.0  2616.0
2021     941.0  1667.0
2023    2141.0  3671.0
2024    1738.0  2634.0
2025    2201.0  3158.0

Evolution of mean and median times

In [23]:
mean_median_by_year = df.groupby("year")["finish_seconds"].agg(["mean", "median"])
print(mean_median_by_year)

             mean  median
year                     
2010  2175.604348  2133.5
2011  2283.334078  2219.0
2012  2385.015192  2363.0
2014  2178.928636  2144.0
2015  2578.058213  2574.0
2016  2147.951908  2155.0
2017  2438.208919  2434.0
2018  2483.670350  2486.0
2019  2759.439222  2753.0
2021  2562.664368  2505.0
2023  2716.662369  2629.0
2024  2982.065401  2883.0
2025  3015.156168  2905.0


Minimum and maximum times over the years

In [24]:
min_max_by_year = df.groupby("year")["finish_seconds"].agg(["min", "max"])
print(min_max_by_year)


         min     max
year                
2010  1430.0  3307.0
2011  1451.0  3496.0
2012  1469.0  4183.0
2014  1255.0  3850.0
2015  1459.0  3954.0
2016  1083.0  3508.0
2017  1262.0  3930.0
2018  1254.0  4693.0
2019  1311.0  5112.0
2021  1302.0  4991.0
2023  1260.0  5678.0
2024  1344.0  5918.0
2025  1387.0  6197.0


# Top Performers 


Identify top performers by average time across all races they participated in



In [25]:
top_runners = (
    df.groupby("runner_name")["finish_seconds"]
    .mean()
    .sort_values()
    .head(10)
)

top_runners


runner_name
MANUEL ANGEL PENAS BLANCO    1083.0
NICOLÁS DOMINGUEZ GARCÍA     1191.0
PEDRO ÁLVAREZ                1215.0
ABDELKADER BOUGUERROUJ       1255.0
RUBEN LIÑARES ARAN           1260.0
IVAN RAÑA FUENTES            1279.0
BORJA COBELO GARCIA          1289.0
SIN DATOS DEL DORSAL 4462    1295.0
LUIS PÉREZ VARELA            1301.6
NASR BOUCHAIB                1302.0
Name: finish_seconds, dtype: float64

# Deliverables 

Summary report 

En esta fase del proyecto ("Data Analytics") se analizaron los datos de la carrera San Silvestre con el objetivo de extraer información relevante sobre el rendimiento de los participantes y la evolución del evento a lo largo del tiempo. Se procesaron 39,862 registros obtenidos mediante técnicas de web scraping. El objetivo ha sido transformar datos brutos en insights sobre el rendimiento deportivo y la evolución social de una de las carreras más emblemáticas de la ciudad.

El análisis de los tiempos medios por grupos de edad muestra diferencias claras entre categorías. Los grupos Senior Masculino (2.049s) y Veteranos B Masculino (2.061s) presentan los mejores tiempos, mientras que categorías con limitaciones físicas o de mayor edad (como DINTF, 4.524s o VTDF, 3.554s) registran tiempos más elevados, siguiendo patrones esperables en pruebas populares. Por género, los participantes masculinos presentan, de media, tiempos de finalización inferiores a los femeninos (2.444s vs 2.982s).

En cuanto a la evolución temporal, la participación aumenta entre 2018 y 2019, desciende notablemente en 2021 (probablemente como consecuencia de la pandemia) y se recupera con fuerza a partir de 2023, alcanzando los valores más altos del periodo analizado. Aunque la participación masculina es mayor en todos los años, la participación femenina muestra una tendencia creciente, lo que indica una progresiva reducción de la brecha de género.

El estudio de los tiempos de finalización revela un incremento gradual de la media y la mediana a lo largo de los años, acompañado de una estabilidad en los tiempos mínimos y un aumento progresivo de los tiempos máximos. Esto sugiere que, aunque el nivel de los corredores más rápidos se mantiene constante, el evento ha incorporado un mayor número de participantes recreativos.

Finalmente, se identificaron los corredores con mejor rendimiento, destacando figuras como Manuel Ángel Penas Blanco con una marca récord de 1.083 segundos. La regularidad se confirma como un factor clave del alto rendimiento en el dataset.

El análisis confirma que la San Silvestre es una prueba en pleno crecimiento, cada vez más inclusiva y paritaria, con patrones de rendimiento que reflejan perfectamente su transición de carrera competitiva a gran festival ciudadano.